vicodyn/src/gateway/vicodyn.cpp

```diff
@@ -201,7 +203,8 @@ auto vicodyn_t::consume(const std::string& uuid,
     wrapped_gateway->consume(uuid, name, version, endpoints, protocol, extra);
     mapping.apply([&](proxy_map_t& mapping){
         if(name == "node") {
-            peers.register_peer(uuid, endpoints, extra);
+            auto peer = peers.register_peer(uuid, endpoints, extra);
+            darkmetrics.subscribe(peer->x_cocaine_cluster());
```

Comment:
Is it any way to unsubscribe listener from cluster? Is it needed?

Reply:
It's very expensive to detect and seems like it is not needed.